acceptance/tests/connect/local_rate_limit_test.go

```diff
@@ -89,22 +101,21 @@ func assertRateLimits(t *testing.T, opts *assertRateLimitOptions, curlArgs ...st
 	retry.RunWith(opts.retryTimer, t, func(r *retry.R) {
 		// Make up to the allowed numbers of calls in a second
 		t0 := time.Now()
-		for i := 0; i < opts.rps; i++ {
-			output, err := k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, args...)
-			require.NoError(r, err)
-			require.Contains(r, output, opts.successOutput)
-		}
+
+		output, err := k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, append(args, repeatAddr)...)
+		require.NoError(r, err)
+		require.Contains(r, output, opts.successOutput)
 
 		// Exceed the configured rate limit.
-		output, err := k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, args...)
+		output, err = k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, append(args, addr)...)
+		require.True(r, time.Since(t0) < time.Second, "failed to make all requests within one second window")
```

Comment:
failing fast(er) here, since if the requests didn't happen in 1s the below assertions are not helpful in debugging the test